Sources/MapboxMaps/Foundation/MapboxMap.swift

```diff
@@ -511,6 +511,66 @@ extension MapboxMap: AttributionDataSource {
     }
 }
 
+// MARK: - Feature State -
+
+extension MapboxMap {
+
+    /// Update the state map of a feature within a style source.
+    /// Update entries in the state map of a given feature within a style source. Only entries listed in the state map
+    /// will be updated. An entry in the feature state map that is not listed in `state` will retain its previous value.
+    ///
+    /// - Note: updates to feature state are asynchronous, so changes made by this method migth
+    ///         not be immediately visible using getStateFeature().
+    ///
+    /// - Parameters:
+    ///   - sourceId: Style source identifier
+    ///   - sourceLayerId: Style source layer identifier (for multi-layer sources such as vector sources). Defaults to `nil`.
+    ///   - featureId: Identifier of the feature whose state should be updated
+    ///   - state: Map of entries to update with their respective new values
+    public func setFeatureState(sourceId: String, sourceLayerId: String? = nil, featureId: String, state: [String: Any]) {
+        __map.setFeatureStateForSourceId(sourceId,
+                                         sourceLayerId: sourceLayerId,
+                                         featureId: featureId,
+                                         state: state)
+    }
+
+    /// Get the state map of a feature within a style source.
+    ///
+    /// - Note: Updates to feature state are asynchronous, so changes made by other methods might not be immediately visible.
```

Comment:
```suggestion
    /// - Note: Updates to feature state are asynchronous, so changes made by other methods might not be immediately visible.
```

So what happens if I set and then get? Since get returns its result asynchronously, is it guaranteed to be queued after the set is complete?

Reply:
It doesn't make any guarantees here from what i can tell. This seems like the intended design of this API though. @pengdev do you have any additional context on the use case this API is targeting?